base/task.jl

```diff
@@ -767,7 +767,7 @@ end
 
 function enq_work(t::Task)
     (t._state === task_state_runnable && t.queue === nothing) || error("schedule: Task not runnable")
-    if t.sticky || Threads.threadpoolsize() == 1
+    if t.sticky || Threads.maxthreadid() == 1
         tid = Threads.threadid(t)
         if tid == 0
```

Comment:
Alternatively, simply this:
```suggestion
        tid = t.sticky ? Threads.threadid(t) : 1
        if tid == 0
```

Reply:
Note that the same change may be need in other `_wait2` functions in this file

Reply:
I don't think so; can you verify?